src/pint/toa.py

```diff
@@ -144,12 +145,43 @@ def get_TOAs(
     if usepickle:
         picklefile = _check_pickle(timfile)
         if picklefile:
-            timfile = picklefile
+            t = TOAs(picklefile)
+            if (
+                include_gps is not None
+                and t.clock_corr_info.get("include_gps", None) != include_gps
+            ):
+                log.info("Pickle contains wrong include_gps")
+                updatepickle = True
+            if (
+                include_bipm is not None
+                and t.clock_corr_info.get("include_bipm", None) != include_bipm
+            ):
+                log.info("Pickle contains wrong include_bipm")
+                updatepickle = True
+            if (
+                bipm_version is not None
+                and t.clock_corr_info.get("bipm_version", None) != bipm_version
+            ):
+                log.info("Pickle contains wrong bipm_version")
+                updatepickle = True
+            if ephem is not None and t.ephem != ephem:
+                log.info("Pickle contains wrong ephem")
+                updatepickle = True
+            if planets is not None and t.planets != planets:
+                log.info("Pickle contains wrong planets")
+                updatepickle = True
         else:
             # Pickle either did not exist or is out of date
             updatepickle = True
-    t = TOAs(timfile)
+    if not usepickle or updatepickle:
```

Comment:
Note that we could be finer grained than this, which would help speed things up.  For instance, if a new ephem is used, all we need to do is re-compute the important table columns, not completely re-process the text .tim file.  The same applies to (I think) all of the other keyword arguments.

Reply:
Do we have statistics on which of these operations (parsing the text file, applying clock corrections, or computing Earth positions and velocities) are slow? The later operations also depend on the result of the earlier operations, I think?

Reply:
Yeah, you are right about some of them depending on earlier operations.  But I think we just need to be careful with that.  And I'm not sure that we have the detailed info on the timing of each step in get_TOAs() handy.  But it would be pretty easy to get.  I'm pretty sure that the posvels are the most costly.  And the text .tim file reading and parsing is costly too.  But I'd have to poke around a bit.  It is possible that @paulray might have the more information from the profiling that his student did this past summer (and which is now in the PINT paper).  I'll tag Chloe here as well in case she pays attention to this stuff still!  @champagne-cmd

Reply:
Clock corrections are kind of a pain - un-applying one and applying a different one requires some non-trivial code. The other two are easy.